src-built-in/components/UIAPI/hooks/linker.ts

```diff
@@ -1,24 +1,120 @@
+        finsembleWindow.hide();
+    };
+
+    const focusWindow = () => {
+        finsembleWindow.focus();
+    };
+
+    const setInitialChannels = () => {
+        const allChannels: any = FSBL.Clients.LinkerClient.getAllChannels();
+        dispatch(actions.SET_CHANNELS({
+            channels: allChannels,
+        }));
+    };
+
+    /** If user switches the linker channel for different components, this function would be invoked. It will dispatch another
+     *  action to update the linker's state according to the linker setup on the switched component.
+     *
+     * @param err  error from FSBL
+     * @param msg  msg from FSBL
+     */
+    const onActiveChannelsChanged = (err: any, msg: any) => {
+        if (err) FSBL.Clients.Logger.system.error(`Failed to update the linker state.`);
+        dispatch(actions.UPDATE_ACTIVE_CHANNELS({
+            channelNames: msg.data.channels,
+            windowIdentifier: msg.data.windowIdentifier,
+        }));
+        msg.sendQueryResponse(null, null);
+    }
+
+    const initAccessibleLinkerMode = () => {
+        const accessibilityCallback = (err: any, value: boolean) => {
+            if (err) FSBL.Clients.Logger.system.error(`Failed to get accessibleLinker value: ${err}`);
+            dispatch(actions.SET_ACCESSIBILITY({
+                isAccessibleLinker: !!value,
+            }));
+            FSBL.Clients.ConfigClient.getValue("finsemble.accessibleLinker", accessibilityCallback);
+        }
+    }
 
     const toggleChannel = (channelId: number) => {
-        dispatch(actions.toggleChannel(channelId));
+        const channelName = state.channels[channelId].name;
+        const channelActive = state.channels[channelId].active;
+        const windowIdentifier = state.windowIdentifier;
+
+        const linkCallback = (isActive: boolean) => {
+            return (err:any) => {
+                if (err) {
+                    return FSBL.Clients.Logger.error('Error in linkerHook, linkCallback', err);
+                }
+
+                dispatch(actions.UPDATE_CHANNEL_STATUS({
+                    channelId,
+                    active: isActive
+                }));
+            };
+        };
+
+        const numberOfActiveChannels = Object.values(state.channels)
+            .map(c => c.active)
+            .reduce((a, c) => {
+                return a + c;
+            });
+
+        // We're activating if the channel is currently inactive
+        const activating = !channelActive;
+        const deactivating = channelActive;
+        const allowedToActivateAnotherChannel = numberOfActiveChannels < MAXIMUM_ALLOWED_ACTIVE_CHANNELS;
+        // if we're toggling a channel to be inactive OR
+        // we're trying to toggle one to be active AND we have not hit our maximum,
+        // let the call go through. Otherwise log an error.
+        if (deactivating) {
+            FSBL.Clients.LinkerClient.unlinkFromChannel(channelName, windowIdentifier, linkCallback(false));
```

Comment:
I'd suggest renaming linkerCallback to something like ```updateActivationTo(false/true)```. It's more readable and people would know what that function is doing without looking into it

Reply:
https://github.com/ChartIQ/finsemble-ui/pull/1/commits/4cef90004bdd6de35b8c8274c024e3a31ab076b1

If this is good with you, i'll close out this PR and move the card. Thanks!